### PR TITLE
fix various typos and errors in the comment.

### LINKED ISSUE #3459: Error comment in LogSub
See the code
https://github.com/kaldi-asr/kaldi/blob/b5385b46c135f8d1f2bf4f9924e287e6fd91cd23/src/base/kaldi-math.h#L226-L227

It actually returns `log(exp(x) - exp(y))`

### PR DESCRIPTION
**Pull request by csukuangfj**
Fix #3459

### CODE REVIEW

**Comment by csukuangfj** [comment]
@danpovey 

I've collected lots of changes. Please have a review.

**Comment by danpovey** [comment]
@desh2608 can you please check through these and comment if you see anything that doesn't look right?  If you say it's OK to merge, I'll merge.

File: src/fstext/kaldi-fst-io.h [hunk 1]
```diff
@@ -87,7 +87,7 @@ fst::VectorFst<fst::StdArc> *ReadAndPrepareLmFst(std::string rxfilename);
 
 // This is a Holder class with T = VectorFst<Arc>, that meets the requirements
 // of a Holder class as described in ../util/kaldi-holder.h. This enables us to
-// read/write collections of FSTs indexed by strings, using the Table comcpet (
+// read/write collections of FSTs indexed by strings, using the Table concpet (
```
Comment (desh2608) [inline src/fstext/kaldi-fst-io.h]: Is this a typo?

File: src/lm/mikolov-rnnlm-lib.cc [hunk 2]
```diff
@@ -759,7 +759,7 @@ void CRnnLM::restoreNet() {   // will read whole network structure
     }
   }
   if (filetype == TEXT) {
-    goToDelimiter(':', fi);    // direct conenctions
+    goToDelimiter(':', fi);    // direct connenctions
```
Comment (desh2608) [inline src/lm/mikolov-rnnlm-lib.cc]: typo

File: src/lm/arpa-lm-compiler.h [hunk 3]
```diff
@@ -31,7 +31,7 @@ class ArpaLmCompilerImplInterface;
 
 class ArpaLmCompiler : public ArpaFileParser {
  public:
-  ArpaLmCompiler(ArpaParseOptions options, int sub_eps,
+  ArpaLmCompiler(const ArpaParseOptions& options, int sub_eps,
```
Comment (desh2608) [inline src/lm/arpa-lm-compiler.h]: I'm not sure about the intended effect of changes here (and similar ones elsewhere) for making the variables `const` and pointers. I hope you have got them approved by @danpovey.

File: src/fstext/kaldi-fst-io.h (hunk 1 above)
Comment (csukuangfj) [reply src/fstext/kaldi-fst-io.h]: @desh2608 
yes, you're correct.

File: src/lm/mikolov-rnnlm-lib.cc (hunk 2 above)
Comment (csukuangfj) [reply src/lm/mikolov-rnnlm-lib.cc]: fixed.

File: src/lm/arpa-lm-compiler.h (hunk 3 above)
Comment (csukuangfj) [reply src/lm/arpa-lm-compiler.h]: & ---> to avoid unnecessary copy
const ---> we do not need to change it inside the constructor.

Comment (desh2608) [reply src/lm/arpa-lm-compiler.h]: Ah, I see, a pass by reference for the arguments which will not be changes inside the function.

**Comment by desh2608** [comment]
@danpovey I think this can be merged now.

**Pull request merged**